src/metahyper/utils.py

```diff
@@ -115,6 +117,10 @@ def instance_from_map(
         ValueError: if the request is invalid (not a string if allow_any is False),
             or invalid key.
     """
+    if isinstance(request, MissingDependencyError):
+        # This happens when some optional dependancy is missing, the error
+        # message will signal to the user what to do
+        raise request
```

Comment:
Shouldn't we check for the type of the `mapping[request]` here (or `instance` later on), instead of the `request`, which is a key from the `mapping`? Since we have the mapping: `SurrogateModelMapping = {"deep_gp": DeepGP, ...}`

Reply:
Oh you're absolutely right :) Will fix